### PR TITLE
Added fix_rotation and no_logs options

### PR DESCRIPTION
**Pull request by buscarini**
Not sure if you would be interested in this, but just in case I have added 2 options. No_logs just removes the log file after transcoding. Fix rotation fixes iPhone video rotation with ffmpeg before transcoding. I tried to do this using Handbrake but I couldn't. Handbrake seems to really want things to stay in landscape mode, so a portrait video that transcoded was looking to the left, when applied a 90º clockwise turn ended looking to the right (so it applied a 180º rotation). This way is not ideal as it's slower, but it works for me.

### CODE REVIEW

File: batch.sh [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+usage() {
+	echo "batch.sh path_to_videos"
+	exit 0
+}
+
+input=$1
+script_path="$(cd "$(dirname "$0")" && pwd)"
+
+if [[ -d "$input" ]];
```
Comment (vitorgalvao) [inline batch.sh]: If you’re putting `then` in the next line, then putting a `;` here doesn’t make much sense (it isn’t needed). In addition, you’re separating them here, but on the rest of the submission you keep them together. The rest of the script already in place seems to keep it together, so it should likely be done here as well.

Not that it’ll change the result, naturally, but consistency in these small things is always a good thing<sup>1</sup>.

---

<sup>1</sup>Arguable if the alternative brings a great advantage in that particular instance, without sacrificing readability.

**Comment by donmelton** [comment]
José,

Thank you for your pull request. However, I don't see a need for either of these features.

I take the blame for poorly and incompletely explaining HandBrake's "rotate" filter earlier. My apologies. 

HandBrake, and therefore `transcode-video.sh`, can rotate portrait to landscape and landscape to portait just fine. I've tested this myself now many, many times. (And let me pause to say, "thank you", for giving me the incentive to understand this much better.)

As I mentioned before, to rotate 90 degrees (a clockwise turn), simply use the value `4` in the argument:

    transcode-video.sh --filter rotate=4 "/path/to/Movie.mkv"

What I didn't mention is that you can combine, i.e. add, the different values to achieve other effects. For example, the value `3` is used to cause a vertical and horizontal flip (basically a 180-degree turn). If you combine that with a 90-degree rotation then you get a 270-degree turn. This is done by adding the two values, `3 + 4`, together, i.e. you use the value `7` like this:

    transcode-video.sh --filter rotate=7 "/path/to/Movie.mkv"

One of the `--filter rotate=4` (90 degrees), `--filter rotate=3` (180 degrees) or `--filter rotate=7` (270 degrees) usages should always solve your problem. If they don't, there's something else wrong going on.

So, another rotation feature is redundant.

Re-transcoding the video is not a good way to solve this problem. It's slow and your implementation does it without the benefit of the rate control system implemented in the script itself. This can significantly degrade quality.

Plus, you're adding a dependency on another tool, `ffmpeg`, which I don't want to do. You're also not validating the availability of that tool.

Diagnostic logs from HandBrake are incredibly valuable. Not only for helping me diagnose user problems when they send them via email or post them here, but they essentially document your inputs and outputs. And you can mine them for all sorts of useful information even if you no longer possess the input and output videos. That's why I wrote the `query-handbrake-log.sh` script which is also part of this project -- to provide a way to inspect these logs and demonstrate their value.

So, adding a feature to remove them is at odds with my goals.

But, just as important, this is completely unnecessary in a command line-based tool. There is almost no additional cost in executing a `rm *.log` command after transcoding at the prompt or in your batch script if you truly want to rid yourself of the log file.

So, I also won't add this feature for the same reason I won't ever add an `--output` option -- it's a complication better solved outside the script.

The implementation is much simpler than the other feature, but it's dangerous in the same way. What do I mean? You're calling `rm` on a file in the current directory.

If you ever have to call `rm` in an end-user script, only to that for something you've created in the `/tmp` directory. You always want to avoid destructive file system operations in what is likely to be a user directory.

When submitting a pull request, it's best to break features into separate patches. That way they can be evaluated separately.

And always match the indenting format of the original source. There are no tab characters in my source files. The code in your patch is indented with tabs. This causes numerous alignment problems, especially in the help text.

Also, you're not following my nomenclature. Your two options contain underscores instead of hyphens and they're not in an "action/object" style.

Please don't be discouraged from submitting patches in the future. I'll close this now but feel free to continue commenting here.

**Comment by buscarini** [comment]
First thing, sorry because I knew this didn't really fit a pull request, but I wasn't sure how to send you this in case you were interested (maybe a comment in the issue?). This is just some code I quickly put together to make it work for me.

About the rotate, you explained it well, that wasn't the problem. I modified the script to add the correct rotate parameter to the HandbrakeCLI command. I tried it with a portrait video and the result was either landscape left no rotation, or landscape right with rotate=4). Because of this I assumed that I couldn't do it with Handbrake and I turned to ffmpeg. I have just tried and it works correctly passing the parameter to the transcode script, so I messed something up. So, sorry for bothering you with this, I didn't check it correctly.

About removing the log, I don't totally agree with you on this. I mean that the transcode script is deciding the name of the log file, and creating it, so I think if you want to remove it it should be the transcode script who does it. An external script doesn't necessarily know the final name of the log file, and there could be logs for other things there, so deleting *.log doesn't seem very appropriate. Of course I know I this isn't the best way to do it, it would be much better not creating the log in the first place, but this was faster for what I needed (again, not proper for a pull request).

Thanks for taking time to check this, and for the tips. I don't make that many pull requests, and I hadn't thought of matching the indentation style, but I will in the future.

**Comment by donmelton** [comment]
José,

I see your point about the `.log` file being the responsibility of the `transcode-video.sh` script. However, an external script always knows the name of the `.log` file if it knows the name of the input. My apologies for the bad `rm` example there. I was trying to show how to get rid of many `.log` files.

Anyway, I will now reconsider this as a feature. Mostly due to the strength of your conviction and argument.

If I do add this feature, the implementation will simply be not to create the `.log` file in the first place. Much safer than removing it.

And I'll likely call the option `--no-log` (notice the singular noun) or maybe `--disable-log` for added clarity. (If I'm feeling full of piss and vinegar that day I might call it `--disable-incredibly-useful-log`, but probably not. :)

Thanks again for giving me pause to think. Just as valuable as a patch!

**Comment by buscarini** [comment]
Thank you very much, but please, don't feel pissed off 😊 you just chose such great defaults for encoding that I felt I didn't need to check the logs (though, honestly it would be very funny to have a --disable-incredibly-useful-log parameter).

You are very kind; this is the problem with pull requests, sometimes they might be nice, and sometimes it's like someone broke into your house and moved your furniture because they felt it was better this way.

**Comment by donmelton** [comment]
> ...this is the problem with pull requests, sometimes they might be nice, and sometimes it's like someone broke into your house and moved your furniture because they felt it was better this way.

I was drinking my coffee this morning when I read that and almost spit it up laughing. :) But that really _is_ what's it's like. You nailed it. And if you're sensitive enough to know that then I was right about you being polite.

Anyway, that's also a great rationale for not needing logs. And I'm getting closer and closer to adding your feature. Heck, in a flurry of reasonableness, I might add the `--output` feature too. :)

**Comment by donmelton** [comment]
OK, your feature has landed. But I just called it `--no-log`. :)

**Comment by buscarini** [comment]
Thank you!

**Pull request closed without merging**